src/org/linphone/LinphoneActivity.java

```diff
@@ -292,20 +294,20 @@ public void onClick(DialogInterface dialog, int which) {
 			public void notifyReceived(LinphoneCore lc, LinphoneEvent ev, String eventName, LinphoneContent content) {
 				super.notifyReceived(lc, ev, eventName, content);
 				if (content.getSubtype().equals("simple-message-summary")) {
-					SharedPreferences prefs = getSharedPreferences(getPackageName(), MODE_PRIVATE);
-					int count = prefs.getInt("mwi_count", 0);
-					count++;
-					prefs.edit().putInt("mwi_count", count).commit();
-
-					View resources = findViewById(R.id.chat);
-					if (resources != null) {
-						View mwi_badge = resources.findViewById(R.id.mwi_badge);
-						if (mwi_badge != null) {
-							TextView notificationCountTextView = (TextView) resources.findViewById(R.id.textView);
-							if (notificationCountTextView != null) {
-								reloadMwiCount();
-							}
+					try {
+						SharedPreferences prefs = getSharedPreferences(getPackageName(), MODE_PRIVATE);
+						String key = "Messages-Waiting: ";
+						String data = new String(content.getData());
+						data = data.substring(data.indexOf(key) + key.length());
+						int number = Integer.parseInt(data.substring(0, data.indexOf("\r\n")));
+
+						prefs.edit().putInt("mwi_count", number).commit();
```

Comment:
@zackm0571, this looks like the mwi_count is being parsed out of the Messages-Waiting: line in the body of the MWI message. This is correct behavior. It is not simply incremented anymore.